File: src/test/testData/java/implicit/PrimitiveArray.java
package ru.ustinov;

import java.net.URL;
import java.util.HashSet;

class Simple {
    public static void main() {
        <warning descr="This data structure can probably implicitly invoke URL#hashCode()">URL[] a;</warning>
        <warning descr="This data structure can probably implicitly invoke URL#hashCode()">java.net.URL[] b;</warning>
        Integer[] c;
        <warning descr="This data structure can probably implicitly invoke URL#hashCode()">HashSet<URL>[] d, g;</warning>
        HashSet<Integer>[] e, f;
    }
}

File: src/test/testData/kotlin/implicit/PrimitiveArray.kt
import java.net.URL
import java.util.HashSet

@Suppress("UNUSED_VARIABLE")
fun main() {
    <warning descr="This data structure can probably implicitly invoke URL#hashCode()">val a: Array<URL></warning>
    val b: Array<Int>
    <warning descr="This data structure can probably implicitly invoke URL#hashCode()">val c: Array<HashSet<URL>></warning>
    val d: Array<HashSet<Int>>
}